clients/src/main/java/org/apache/kafka/clients/admin/TopicCollection.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.Collection;
+
+public class TopicCollection {
+
+    private Collection<String> topicNames;
+    private Collection<Uuid> topicIds;
+    private final TopicAttribute attribute;
+
+    public enum TopicAttribute {
+        TOPIC_NAME, TOPIC_ID
```

Comment:
Another way to do this might be to make `TopicCollect` abstract and create two implementations:
```java
abstract class TopicCollection
class TopicIdCollection extends TopicCollection
class TopicNameCollection extends TopicCollection

public static TopicIdCollection ofTopicIds(Collection<Uuid> topicIds);
public static TopicNameCollection ofTopicNames(Collection<String> topicNames);
```

Reply:
I originally had something like this but it got tricky when writing the code. I was a little wary of converting between the subtype and the super type in the methods as there were quite a few java warning.

Reply:
That is an option. What is the benefit you see with this way?

Reply:
Mainly better type safety. It feels a bit clumsy in the current implementation that we throw UnsupportedOperationException for some methods depending on the type.

Reply:
We still have to check the type of the class yes? What happens if someone extends the abstract class to another subtype?

Reply:
I guess this prevents the unsupported operation exception in the methods themselves though. We would need to convert the TopicCollection to TopicNameCollection before they could access the collection of names though, so I'd suspect we'd still need to handle there.

Reply:
Recent Java versions make it possible to seal a class so that you can control who can create a subclass. Until then, we can document that subclassing is not supported.

Reply:
Is there a good way to do this for the result class? I'm not sure if it works to make the already existing class abstract. Or should we just keep as is?